### ISSUE TITLE
More doctest examples for Linear Algebra standard library

### ISSUE THREAD

**Issue by kshyatt** [JuliaLang/julia#18799]
http://docs.julialang.org/en/latest/stdlib/linalg/

Many of these functions are confusing and a simple, explanatory example would go far.

See the docs for [cross](http://docs.julialang.org/en/latest/stdlib/linalg/#Base.cross) for an example of what a doctest can look like.

See the [docs about docs](http://docs.julialang.org/en/latest/manual/documentation/) for more information about how to write doctests.

Before you submit a PR, please make sure to run `make docs && make -C doc doctest` and confirm that all doctests pass!

**Comment by SaschaMann** [comment]
I have started working on this issue. I'm somewhat new to Julia, so please excuse it if the following questions are "stupid":

Is there a way to run `make -C doc doctest` _just_ for the linalg docs for more convenient inbetween checks?

I came across some functions with examples as normal syntax-highlighted code. For example [Bidiagonal](http://docs.julialang.org/en/latest/stdlib/linalg/#Base.Bidiagonal). Is there a particular reason why these aren't doctests or can/should I just rewrite them to doctests?

**Comment by kshyatt** [comment]
@SaschaMann welcome to Julia! Please don't feel bad about asking questions.

> Is there a way to run `make -C doc doctest` just for the linalg docs for more convenient inbetween checks?

No. Changes in `linalg` may break tests in the manual (this has happened to me!) so it's best to run the full test suite.

> Is there a particular reason why these aren't doctests or can/should I just rewrite them to doctests?

I wrote that example. We don't want to use `rand` in doctests because it will be different from run to run. You could very easily rewrite that example as a doctest using a prespecified vector, like `a = [1, 2, 3, 4]`.

Hope that helps!

**Comment by SaschaMann** [comment]
There are still plenty of functions that would benefit from an example but I don't understand many of them well enough to provide a helpful example.

**Comment by kshyatt** [comment]
That's OK Sacha. Even very small improvements are helpful. If you'd like to
ask for resources to understand what the functions do I could give you some
suggestions?

**Comment by Sacha0** [comment]
> If you'd like to ask for resources to understand what the functions do I could give you some suggestions?

Please feel welcome to ping me with questions as well. Best and welcome!

**Comment by SaschaMann** [comment]
So far I've only used examples that I either came up with myself or had in my class notes; if I were to take an example from a book, because it has results that make it easier to grasp than the results of random matrices or 1 2 3 4 etc, would I cite it with `[^A12]` where `A` is the first letter of the last name and `12` the year of publishing and then `[^A12]: source`? Not sure I got the naming scheme correctly.

@kshyatt If you have a recommendation for something that explains the different factorization and decomposition methods used in the package (in general, not necessarily Julia specific) that would be nice. I can figure out the Julia functions from there. Unfortunately the Wikipedia articles for these are often not quite enough.


Thanks to both of you for your help and friendliness towards newbies! :)
